misk-action-scopes/src/main/kotlin/misk/scope/ActionScope.kt

```diff
@@ -83,7 +85,15 @@ class ActionScope @Inject internal constructor(
   }
 
   /** Returns true if currently in the scope */
-  fun inScope(): Boolean = threadLocalScope.get() != null
+  override fun inScope(): Boolean = threadLocalScope.get() != null
+
+  override fun <T> get(type: Class<T>): T? {
+    if (!inScope()) {
+      return null
+    }
+
+    return get(Key.get(type))
```

Comment:
As it is today, this class has dependency on Guice since it has Guice providers as property which are used when a Key is given.  I wonder if we want to remove this dependency and use a more default Supplier interface.

Reply:
For now, we are keeping the Guice module in misk-api.